### PR TITLE
[Jax][Qwen] Several fixes for `Qwen3-4B-2507-Instruct-FP8` in jax-path

### PR DESCRIPTION
**Pull request by lk-chen**
# Description

`Qwen3-4B-2507-Instruct-FP8` is the most achievable model to test FP8 linear for #1623 (for now, because #1633 and #1640 enabled block-wise FP8 linear method, and we are missing FP8-RmsNorm and/or FP8-MoE). This PR contains several fixes for this model:

1. set `dtype=model.dtype`. It's meaningless to set `param_dtype`, because we should [keep the original type from checkpoint](https://github.com/vllm-project/tpu-inference/blob/91d7109627bc968fff535fa223ce29b7a8d302bb/tpu_inference/models/jax/utils/weight_utils.py#L756-L758). Otherwise we load an FP8 checkpoint onto the device as floating type, that loses the whole point of quantization. This PR sets `dtype` of each layer to follow [upstream definition](https://docs.vllm.ai/en/v0.15.1/configuration/engine_args/#-dtype).

2. considers skipped layers, following upstream code (see code comment for details). Model files are updated to pass `prefix` of each module.

3. use `jax.set_mesh` when necessary.

4. Known issue 
  * #1684 
  * #1661 

# Tests

1. CI

2. benchmark and eval
```
# start server
export CURRENT_VLLM_MODEL=Qwen/Qwen3-4B-Instruct-2507-FP8
MODEL_IMPL_TYPE=flax_nnx vllm serve $CURRENT_VLLM_MODEL \
    --max-model-len=8192 \
    --disable-log-requests \
    --no-enable-prefix-caching

# benchmark
export CURRENT_VLLM_MODEL=Qwen/Qwen3-4B-Instruct-2507-FP8
vllm bench serve \
    --model $CURRENT_VLLM_MODEL \
    --num-prompts 1000 \
    --dataset-name random \
    --random-input-len 1024 \
    --random-output-len 4096

# eval
export CURRENT_VLLM_MODEL=Qwen/Qwen3-4B-Instruct-2507-FP8
lm_eval \
    --model local-completions \
    --model_args base_url=http://localhost:8000/v1/completions,model=$CURRENT_VLLM_MODEL \
    --tasks gsm8k_cot \
    --batch_size auto \
    --apply_chat_template \
    --num_fewshot 8
```
result
```
============ Serving Benchmark Result ============
Successful requests:                     1000      
Failed requests:                         0         
Benchmark duration (s):                  1641.95   
Total input tokens:                      1024000   
Total generated tokens:                  4096000   
Request throughput (req/s):              0.61      
Output token throughput (tok/s):         2494.60   
Peak output token throughput (tok/s):    3447.00   
Peak concurrent requests:                1000.00   
Total token throughput (tok/s):          3118.25   
---------------Time to First Token----------------
Mean TTFT (ms):                          615918.11 
Median TTFT (ms):                        597434.32 
P99 TTFT (ms):                           1339273.85
-----Time per Output Token (excl. 1st token)------
Mean TPOT (ms):                          83.08     
Median TPOT (ms):                        72.91     
P99 TPOT (ms):                           122.62    
---------------Inter-token Latency----------------
Mean ITL (ms):                           83.08     
Median ITL (ms):                         66.09     
P99 ITL (ms):                            123.75    
==================================================
```
|  Tasks  |Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|---------|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k_cot|      3|flexible-extract|     8|exact_match|↑  |0.7718|±  |0.0116|
|         |       |strict-match    |     8|exact_match|↑  |0.0955|±  |0.0081|

Compare with torchax path: #1661 

# Checklist

Before submitting this PR, please make sure:
- I have performed a self-review of my code.
- I have necessary comments in my code, particularly in hard-to-understand areas.
- I have made or will make corresponding changes to any relevant documentation.

### CODE REVIEW

**Comment by vanbasten23** [comment]
@lk-chen This PR broke the lora test, for example [nightly](https://screenshot.googleplex.com/zmjxkAjsr2ib9pz) (you can verify by "VLLM_LOG_LEVEL=INFO VLLM_XLA_CHECK_RECOMPILATION=1 VLLM_USE_V1=1 MODEL_IMPL_TYPE=vllm TPU_BACKEND_TYPE=jax pytest -s -vv tests/lora/test_lora.py -k single"). It fails because of recompilation. Can you take a look? Thanks.
